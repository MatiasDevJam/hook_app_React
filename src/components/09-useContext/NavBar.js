import React from 'react';
import { NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="nav">
            
            <NavLink exact activeClassName="active" to="/" className="nav-link" >Home</NavLink>
            <NavLink exact activeClassName="active" to="/login" className="nav-link" >Login</NavLink>
            <NavLink exact activeClassName="active" to="/about" className="nav-link" >About</NavLink>
            
        </nav>
    )
}
