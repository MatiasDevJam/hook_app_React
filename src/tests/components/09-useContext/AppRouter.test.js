import React from 'react'
import { mount } from 'enzyme'
import { UserContext } from '../../../components/09-useContext/UserContext'
import { AppRouter } from '../../../components/09-useContext/AppRouter'

describe('Prueba en <AppRouter />', () => {

    const user = {
        id: 1,
        user: 'Julieta'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter />
        </UserContext.Provider>
    )

    test('debe mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })
    
    
})
