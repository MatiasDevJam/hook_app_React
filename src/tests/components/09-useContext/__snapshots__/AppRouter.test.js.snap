// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <AppRouter /> debe mostrarse correctamente 1`] = `
<div>
  <nav
    className="nav"
  >
    <a
      activeClassName="active"
      aria-current="page"
      className="nav-link active"
      exact={true}
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      activeClassName="active"
      className="nav-link"
      exact={true}
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <a
      activeClassName="active"
      className="nav-link"
      exact={true}
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "user": "Julieta"
}
      </pre>
    </div>
  </div>
</div>
`;
